pkgs/development/python-modules/ninja/stub/ninja/__init__.py

```diff
@@ -0,0 +1,8 @@
+import subprocess
+import sys
+
+def _program(name, args):
+    return subprocess.call([name] + args, close_fds=False)
+
+def ninja():
+    raise SystemExit(_program('ninja', sys.argv[1:]))
```

Comment:
Should we hardcode the path to the ninja executable so we get a proper runtime dependency

Reply:
I'm a little out of my depth with this. With normal derivations, I'm under impression that including in `buildInputs` is preferred since it's structured (maybe helps with cross too?) and will be added to PATH. With Python derivations, I don't know how it works exactly, but I had to add it to propagatedBuildInputs to achieve the same effect. What's the difference between doing that and hardcoding?

Reply:
I thought about it a bit more and like the idea of hardcoding the path in the function. I had to keep `ninja` also in propagatedBuildInputs because some users of this package do expect ninja to just be in PATH.

Reply:
With regular derivations we do not need to explicitly hardcode because references get compiled in. With Python that does not happen so we need to explicitly hardcode.